jms/runtime/src/main/java/io/quarkus/artemis/jms/runtime/ArtemisJmsRecorder.java

```diff
@@ -13,20 +14,15 @@
 @Recorder
 public class ArtemisJmsRecorder {
 
-    public ArtemisJmsWrapper getDefaultWrapper() {
-        return new ArtemisJmsWrapper() {
-            @Override
-            public ConnectionFactory wrapConnectionFactory(ActiveMQConnectionFactory cf) {
-                return cf;
-            }
-        };
+    public Function<ConnectionFactory, Object> getDefaultWrapper() {
+        return cf -> cf;
```

Comment:
I thought Quarkus has a policy of using anonymous classes instead of lambda because of performance reasons.

Reply:
@middagj See [this zulip chat](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/lambdas.20in.20extensions). So we can use lambdas.

Reply:
Great, I like the lambda notation better. I was planning to do the research myself as I mentioned it before (after got corrected on in https://github.com/quarkusio/quarkus/pull/15350#discussion_r583563115).

Makes sense, that the JVM optimizes these things by itself instead of relying on devs (same as with StringBuilder vs concat operator previously (nowadays it is even faster using invokedynamic)).

Reply:
@middagj can this thread then be resolved?

Reply:
Yes, of course